support/build-debian.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+# Copyright 2024 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+DEB_ARCH="${1}"
+export GOOS="linux"
+case "${DEB_ARCH}" in
+    "armhf")
+        export GOARCH="arm"
+        export GOARM="6"
+        ;;
+    "arm64")
+        export GOARCH="arm64"
+        ;;
+    "amd64")
+        export GOARCH="amd64"
+        ;;
+    *)
+        echo "Usage: $0 <DEBARCH>" 1>&2
+        exit 1
+        ;;
+esac
+VERSION="$(git describe --tags --dirty)"
+TMPDIR="$(mktemp --tmpdir -d "nt-connect-XXXXX")"
+trap 'rm -rf $TMPDIR' EXIT
+export DESTDIR="${TMPDIR}/nt-connect-${VERSION#v}_${GOOS}_${DEB_ARCH}"
+mkdir -p "${DESTDIR}/DEBIAN"
+make install
+cat > "${DESTDIR}/DEBIAN/control" <<EOF
+Package: nt-connect
+Version: ${VERSION#v}
+Architecture: any
```

Comment:
This doesn't seem to work,
```
root@9a16e99a3eda:/# dpkg --install /dist/nt-connect-0.0.1_linux_amd64.deb 
dpkg: error processing archive /dist/nt-connect-0.0.1_linux_amd64.deb (--install):
 package architecture (any) does not match system (amd64)
Errors were encountered while processing:
 /dist/nt-connect-0.0.1_linux_amd64.deb
 ```

but I don't know why. I've never used `dpkg-deb` directly so I suspect that the low level tool does not handle the translation.

In mender we use `dpkg-buildpackage` but for that you need to follow more of the Debian policy.

Reply:
Oops, this was also a follow-error from previous attempts to build with `dpkg-buildpackage`. The Architecture should be set to  the build target.

Reply:
This should be fixed now.